### ISSUE TITLE
Dropdown positioned to far down when container is within relative div within a table row

### ISSUE THREAD

**Issue by Loadmaster** [select2/select2#4382]
## Environment
Select2 4.0.2.
Seen on desktop Firefox, Chrome, and Android browsers.

## Problem
For a container element that has `position:relative` with `multiple` and `tags:true` enabled, and `closeOnSelect:false` disabled.

When entering a search pattern, if there is enough room in the visible scroll area below the container for the drop-down, the drop-down is supposed to be positioned immediately below the container; however, it is being positioned too far down on the page below the container. It has the correct `left` position, but not the correct `top` position.

Further research shows that the `AttachBody.prototype._positionDropdown()` method appears to be assuming `static` positioning of the parent container, so it is adding in the parent's Y coordinate instead of assuming an `absolute` position offset of zero.

I don't know why this does not show up in other places (e.g., the Select2 examples page, which appears to work fine on Android browsers), except perhaps for the fact that my `<select>` element is contained within a `<td>` table element.

## Quick Fix
Change the existing code (circa line 4230) from:
```
    if (newDirection == 'above' ||
      (isCurrentlyAbove && newDirection !== 'below')) {
      css.top = container.top - dropdown.height;
    }
```
to this:
```
    if (newDirection == 'above' ||
      (isCurrentlyAbove && newDirection !== 'below')) {
      css.top = -dropdown.height;    // FIX: Ignore outer container location
    }
```
This seems to fix the problem.

However, I am not confident that this will not have adverse effects on existing applications, specifically for drop-downs contained within `static` positioned parent elements. Perhaps an additional test for non-`static` placement should also be added.

**Comment by kevin-brown** [comment]
This was probably fixed by https://github.com/select2/select2/pull/4267. If using the `master` version doesn't fix the issue, feel free to leave a new comment with a jsbin that shows the issue.

**Comment by kevin-brown** [comment]
Note that we finally rolled out Select2 4.0.3 last night, which should include this fix.

**Comment by Loadmaster** [comment]
This appears to be fixed in 4.0.3.
Thanks.
